aiida/cmdline/commands/cmd_node.py

```diff
@@ -61,6 +64,86 @@ def repo_ls(node, relative_path, color):
+    required=True,
+)
+@with_dbenv()
+def repo_dump(node, output_directory, force):
+    """Copy the repository files of a node to an output directory."""
+    from aiida.orm.utils.repository import FileType
+
+    output_directory = pathlib.Path(output_directory)
+
+    output_directory.mkdir(exist_ok=True)
+
+    def _copy_tree(key, output_dir):  # pylint: disable=too-many-branches
+        if force:
+            prompt_res = 'r'
+        for file in node.list_objects(key=key):
+            if not key:
+                file_path = file.name
+            else:
+                file_path = key + '/' + file.name
+            if file.type == FileType.DIRECTORY:
+                new_out_dir = output_dir / file.name
+                if new_out_dir.exists():
+                    if not force:
+                        prompt_res = click.prompt(
+                            "Directory '{}' exists. Abort [a], "
+                            'skip [s], merge contents [m], '
+                            'or replace [r]?'.format(new_out_dir),
+                            default='a',
+                            type=click.Choice(['a', 's', 'm', 'r'])
+                        )
+                    if prompt_res == 'a':
+                        raise click.Abort
+                    elif prompt_res == 's':
+                        continue
+                    elif prompt_res == 'r':
+                        # Explicit 'str' cast is needed only in python <=3.5,
+                        # because 'shutil' does not understand pathlib
+                        # objects. This can be removed when Python3.5
+                        # support is dropped.
+                        shutil.rmtree(str(new_out_dir))
+                        new_out_dir.mkdir()
+                    else:
+                        assert prompt_res == 'm'
+                else:
+                    new_out_dir.mkdir()
+
+                _copy_tree(key=file_path, output_dir=new_out_dir)
+            else:
+                assert file.type == FileType.FILE
+                out_file_path = (output_dir / file.name)
+                if out_file_path.exists():
+                    if not force:
+                        prompt_res = click.prompt(
+                            "File '{}' exists. Abort [a], skip [s], or replace [r]?".format(out_file_path),
+                            default='a',
+                            type=click.Choice(['a', 's', 'r'])
+                        )
+
+                    if prompt_res == 'a':
+                        raise click.Abort
+                    elif prompt_res == 's':
+                        continue
+                    else:
+                        assert prompt_res == 'r'
+                        out_file_path.unlink()
+                with node.open(file_path, 'rb') as in_file:
+                    with out_file_path.open('wb') as out_file:
+                        out_file.write(in_file.read())
```

Comment:
does `read()` first load the whole `in_file` into memory?

Reply:
fixed